addons/payment_sips/controllers/main.py

```diff
@@ -17,12 +18,12 @@ class SipsController(http.Controller):
     _notify_url = '/payment/sips/ipn/'
     _return_url = '/payment/sips/dpn/'
 
-    def sips_validate_data(self, **post):
-        sips = request.env['payment.acquirer'].search([('provider', '=', 'sips')], limit=1)
-        security = sips.sudo()._sips_generate_shasign(post)
+    def _sips_validate_data(self, **post):
+        sips = request.env['payment.acquirer'].sudo().search([('provider', '=', 'sips')], limit=1)
```

Comment:
You cannot know which acquirer will be the right one: there could be several acquirers (one per company) sharing the same provider.
Use `tx_sudo = self.env['payment.transaction].sudo()._get_tx_from_feedback_data(...) ; acquirer_sudo = tx_sudo.acquirer_id`